packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -1079,6 +1079,46 @@ class AlignTransition extends AnimatedWidget {
 /// Animated version of a [DefaultTextStyle] that animates the different properties
 /// of its [TextStyle].
 ///
+/// {@tool dartpad --template=stateful_widget_material_ticker}
+///
+/// The following code implements the [DefaultTextStyleTransition]
+///
+/// ```dart
+/// AnimationController _controller;
+///
+/// @override
+/// void initState() {
+///   super.initState();
+///   _controller = AnimationController(
+///     duration: const Duration(seconds: 2),
+///     vsync: this,
+///   )..repeat(reverse: true);
+/// }
+///
+/// @override
+/// void dispose() {
+///   _controller.dispose();
+///   super.dispose();
+/// }
+///
+/// @override
+/// Widget build(BuildContext context) {
+///   return Center(
+///        child: DefaultTextStyleTransition(
+///          style: TextStyleTween(
+///              begin: TextStyle(fontSize: 50, color: Colors.blue, fontWeight: FontWeight.w900),
+///              end: TextStyle(fontSize: 50, color: Colors.red, fontWeight: FontWeight.w100)
+///          ).animate(CurvedAnimation(
+///              parent: _controller,
+///              curve: Curves.elasticInOut
+///          )),
```

Comment:
Instead of building a new animation/tween: These could be stored in instance variables that get re-used.

Reply:
@goderbauer I tried to modify the example as you said. I hope this one would be better.